home/import_helpers.py

```diff
@@ -157,3 +163,55 @@ def extract_errors(data: dict[str | int, Any] | list[str]) -> dict[str, str]:
         pass
 
     return error_message
+
+
+def fix_rows(rows: Generator[dict[str, str], None, None]) -> Iterator[dict[str, str]]:
+    """
+    Fix keys for all rows by lowercasing and removing whitespace
+    """
+    for row in rows:
+        yield fix_row(row)
+
+
+def fix_row(row: dict[str, str]) -> dict[str, str]:
+    return {k.lower().strip(): v.strip() if v else v for k, v in row.items()}
+
+
+def parse_file(
+    file_content: bytes, file_type: str
+) -> Iterator[tuple[int, dict[str, Any]]]:
+    read_rows = read_xlsx if file_type == "XLSX" else read_csv
+    return enumerate(fix_rows(read_rows(file_content)), start=2)
+
+
+def read_csv(file_content: bytes) -> Generator[dict[str, Any], None, None]:
+    try:
+        for row in csv.DictReader(StringIO(file_content.decode())):
+            yield {k.strip().lower(): v for k, v in row.items()}
+    except AttributeError:
+        raise ImportException(
+            "Invalid format. Please check that all row values have headers."
+        )
+
+
+def read_xlsx(file_content: bytes) -> Generator[dict[str, Any], None, None]:
+    workbook = load_workbook(BytesIO(file_content), read_only=True, data_only=True)
+    worksheet = get_active_sheet(workbook)
+
+    def clean_excel_cell(cell_value: str | float | datetime | None) -> str:
+        return str(cell_value).replace("_x000D", "").strip()
+
+    first_row = next(worksheet.iter_rows(max_row=1, values_only=True))
+    header = [clean_excel_cell(cell) if cell else None for cell in first_row]
+
+    for row in worksheet.iter_rows(min_row=2, values_only=True):
+        r = {}
+        if len(row) > len(header):
+            raise ImportException(
+                "Invalid format. Please check that all row values have headers."
+            )
+        for name, cell in zip(header, row):  # noqa: B905
+            if name and cell:
+                r[name.strip().lower()] = clean_excel_cell(cell)
```

Comment:
Same here for the duplicate cleaning of keys. We'll probably still want the excel cell cleaning bit, because that's excel specific